src/Symfony/Component/HttpKernel/Kernel.php

```diff
@@ -440,7 +442,12 @@ protected function build(ContainerBuilder $container)
      */
     protected function getContainerClass()
     {
-        return str_replace('\\', '_', \get_class($this)).ucfirst($this->environment).($this->debug ? 'Debug' : '').'Container';
+        $name = \get_class($this);
+        if (false !== $i = strrpos($name, '\\')) {
+            $name = substr($name, 1 + $i);
+        }
+
+        return $this->name.$name.ucfirst($this->environment).($this->debug ? 'Debug' : '').'Container';
```

Comment:
This change is wrong. We don't want to rely on the name anymore. Right now, name is kind of required only because of the way we tests things internally, but nobody will have the same issue IRL.

Reply:
what's the issue with keeping it in 4.x and removing in 5.0?
fixing tests on lower branches looks even more wrong to me